### PR TITLE
Clean up ExprEvalScheduler::canScheduleCompileTime.

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/scheduler/expr_eval_sched.cpp [hunk 1]
```diff
@@ -19,21 +19,26 @@ bool ExprEvalScheduler::canScheduleCompileTime(Fusion* fusion) {
...
       scheduler_debug_utils::canScheduleRejectReason(
           heuristicType(),
           "Matmul ATen evaluation was disabled by NVFUSER_DISABLE=matmul_expr_eval");
+      return false;
     }
-  } else {
-    scheduler_debug_utils::canScheduleRejectReason(
-        heuristicType(),
-        "Fusion must contain only a single expression of type MatmulOp/LinearOp/SdpaFwdOp/SdpaBwdOp");
+    return true;
```
Comment (Priya2698) [inline csrc/scheduler/expr_eval_sched.cpp:36]: The logic seems inverted here -- `if (!isOptionDisabled(DisableOption::MatmulExprEval))` should return `false`.

File: csrc/scheduler/expr_eval_sched.cpp [hunk 2]
```diff
@@ -19,21 +19,26 @@ bool ExprEvalScheduler::canScheduleCompileTime(Fusion* fusion) {
   if (exprs.size() != 1) {
     scheduler_debug_utils::canScheduleRejectReason(
         heuristicType(), "Fusion must contain only a single expression.");
```
Comment (jacobhinkle) [inline csrc/scheduler/expr_eval_sched.cpp:21]: Out of curiosity, where does the requirement that there is only a single expression come from?

Comment (Priya2698) [reply csrc/scheduler/expr_eval_sched.cpp:21]: This is a remnant of the design choice we made earlier during implementation of `matmul` and `linear`.

We can revisit and lift this restriction -- I don't see any reason why we cannot support it.
Will test with a few cases.

**Comment by wujingyue** [comment]
!build

File: csrc/scheduler/expr_eval_sched.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/scheduler/expr_eval_sched.cpp]: Thanks!

**Review by Priya2698 — APPROVED** [review]
LGTM.

**Pull request merged**